### DISCUSSION TITLE
Support visualizing CompressedPointCloud2

### DISCUSSION THREAD

**Discussion by Aposhian** [foxglove/community#907] [Q&A] [answered]
Recording PointCloud2 messages can create very large MCAP files very quickly. There is some support for easy pointcloud compression in ROS using https://github.com/ros-perception/point_cloud_transport. This uses the type [CompressedPointCloud2](https://github.com/ros-perception/point_cloud_transport_plugins/blob/rolling/point_cloud_interfaces/msg/CompressedPointCloud2.msg) for transport.

Since this is lossless compression, and there is no inter-frame compression, it seems it would be relatively straightforward to add decoding capability to Foxglove Studio, which would allow for saving much smaller MCAPs.

There are three default compression plugins, zlib, zstd, and draco. zlib and zstd are very standard, and draco is built specifically for pointclouds and graphics data. The good news is that Google already has Javascript decoders for Draco: https://github.com/google/draco#WASM-and-JavaScript-Decoders.

**Comment by jtbandes** [comment, marked as answer]
Duplicate of https://github.com/orgs/foxglove/discussions/307
